sled-agent/src/services.rs

```diff
@@ -1419,12 +1419,18 @@ impl ServiceManager {
     }
 
     fn zone_network_setup_install(
-        gw_addr: &Ipv6Addr,
+        gw_addr: Option<&Ipv6Addr>,
         zone: &InstalledZone,
         static_addrs: &[Ipv6Addr],
     ) -> Result<ServiceBuilder, Error> {
         let datalink = zone.get_control_vnic_name();
-        let gateway = &gw_addr.to_string();
+
+        // The switch zone is the only zone that will sometimes have an
+        // unknown underlay address at zone boot on the first scrimlet.
+        let gateway = match gw_addr {
+            Some(addr) => addr.to_string(),
+            None => "unknown".to_string(),
```

Comment:
It looks like we later parse this back into an `Option<Ipv6Addr>` via `parse_gateway` - that seems kinda weird; is there any way we can avoid it? E.g., could we make the `--gateway` argument to zone-setup optional instead of required?

Reply:
Ugh, yeah that definitely looks a bit odd. This was a result of several issues:

- Clap doesn't like when you pass an empty value on a string that takes a value. It always fails even if a flag is not required and it creates confusion because the flag appears to be "not required". By making it "required" a failure is a bit more obvious.
- There is a chance the manifest.xml file could set an empty value on on the `-g` flag `https://github.com/oxidecomputer/omicron/blob/main/smf/zone-network-setup/manifest.xml#L21`. Even though I'm setting it as "unknown" in the property builder, if the properties change for some reason (e.g. if there's a change here https://github.com/oxidecomputer/omicron/blob/main/sled-agent/src/services.rs#L3888 at some point)

That said, yeah, this is definitely ugly. Let me think how I can make it nicer

Reply:
done :)